interface/ShowerParametrization.h

```diff
@@ -8,101 +10,123 @@
+        a0_hadro = transverse_params_hadro.at("a0");
+        a1_hadro = transverse_params_hadro.at("a1");
+        a2_hadro = transverse_params_hadro.at("a2");
+
+        tot_layer_Electro = 28.;
+        tot_layer_Hadro = 52.;
     }
+
     ~ShowerParametrization() {}
 
     // average medium
-    double getRadiationLength() const {return radiationLength_;}
     double getMoliereRadius() const {return moliereRadius_;}
-    double getCriticalEnergy() const {return criticalEnergy_;}
-
-    // longitudinal
-    double meanT(double lny) const {return meant0_+meant1_*lny;}
-    double meanAlpha(double lny) const {return meanalpha0_+meanalpha1_*lny;}
-    double sigmaLnT(double lny) const {return 1./(sigmalnt0_+sigmalnt1_*lny);}
-    double sigmaLnAlpha(double lny) const {return 1./(sigmalnalpha0_+sigmalnalpha1_*lny);}
-    double correlationAlphaT(double lny) const {return corrlnalphalnt0_+corrlnalphalnt1_*lny;}        
-    const std::array<double,28>& getLayerProfile() const {return layerProfile_;}
+    double getInteractionLength() const {return interaction_length_;}
+
+
+    std::array<double,52>& getLayerProfile(int pdgid) {
+        map<int, std::vector<double>>::const_iterator pos = layerProfile_.find(pdgid);
+        if (pos == layerProfile_.end()) {
+            cout << "pdgid doesn't exist"<<endl;
+        } else {
+            int index = 0;
+            typedef std::vector<double>::const_iterator ListIterator;
+            for (ListIterator list_iter = pos->second.begin(); list_iter != pos->second.end(); list_iter++){
+                partLayerProfile_[index] = *list_iter;
+                index++;
+            }
+        }
+        return partLayerProfile_;
+    }
+
     // transversal
-    double r0(int klayer) {return klayer==-1 ? r0layer15_ : (a0_ + a1_*klayer + a2_*klayer*klayer)*r0layer15_/28.;}
+    double r0_electro_(int klayer, int pdgid) {return (a0_electro + a1_electro*klayer + a2_electro*klayer*klayer)*r0layer_Electro_/tot_layer_Electro;}
+    double r0_hadro_(int klayer, int pdgid) { return (a0_hadro + a1_hadro*klayer + a2_hadro*klayer*klayer)/tot_layer_Hadro;}
 
     // fluctuations
-    double spotEnergy() {return alpha_*alpha_;}
-    
-  private:
-  
-    // average medium parameters
-    double radiationLength_; // in cm
-    double moliereRadius_;   // in cm 
-    double criticalEnergy_;  // in MeV 
-
-    // longitudinal parametrisation
-    double meant0_;
-    double meant1_;
-    double meanalpha0_;
-    double meanalpha1_;
-    double sigmalnt0_;
-    double sigmalnt1_;
-    double sigmalnalpha0_;
-    double sigmalnalpha1_;
-    double corrlnalphalnt0_;
-    double corrlnalphalnt1_;
-    std::array<double,28> layerProfile_;
-    
-    // transverse parametrisation
-    double r0layer15_;
-    double a0_;
-    double a1_;
-    double a2_;
-
-    // for fluctuation
-    double alpha_; // the stochastic coefiscient in GeV^1/2
-    
+    std::array<double,3> spotEnergy(int pdgid) {
```

Comment:
Because now on one layer the three different Silicium thickness are implemented and for each of these thickness the energy resolution is different.

Reply:
It would be more easy to understand if you give the silicon thickness as parameter to `spotEnergy()` and return on single number corresponding to this thickness.